### PR TITLE
[SCB-1673] parse the remaining information of RequestMapping

### PR DESCRIPTION
**Pull request by kakulisen**
Some information on the RequestMapping annotation was not processed

### CODE REVIEW

File: oas-generator/oas-generator-core/src/main/java/org/apache/servicecomb/toolkit/generator/context/OasContext.java [hunk 1]
```diff
@@ -47,6 +47,14 @@
 
   private List<ISchemaContext> schemaCtxList = new ArrayList<>();
 
+  private String httpMethod;
+
+  private String[] consumers;
```
Comment (chanjarster) [inline oas-generator/oas-generator-core/src/main/java/org/apache/servicecomb/toolkit/generator/context/OasContext.java]: Typo, should be `consumes`

File: oas-generator/oas-generator-core/src/main/java/org/apache/servicecomb/toolkit/generator/context/OperationContext.java [hunk 2]
```diff
@@ -73,6 +77,10 @@
 
   private String[] consumers;
```
Comment (chanjarster) [inline oas-generator/oas-generator-core/src/main/java/org/apache/servicecomb/toolkit/generator/context/OperationContext.java]: Typo, should be `consumes`

File: oas-generator/oas-generator-core/src/main/java/org/apache/servicecomb/toolkit/generator/context/OperationContext.java [hunk 3]
```diff
@@ -155,6 +165,58 @@ public Operation toOperation() {
     return operation;
   }
 
+  private void processHeaders() {
+
+    if (headers == null) {
+      headers = parentContext.getHeaders();
+    }
```
Comment (chanjarster) [inline oas-generator/oas-generator-core/src/main/java/org/apache/servicecomb/toolkit/generator/context/OperationContext.java]: Provide `getHeaders()` method just like `getHttpMethod()` will be better.

File: oas-generator/oas-generator-core/src/main/java/org/apache/servicecomb/toolkit/generator/context/OperationContext.java [hunk 4]
```diff
@@ -155,6 +165,58 @@ public Operation toOperation() {
...
+        operation.addParametersItem(headerParameter);
+      }
+    });
+  }
+
+  private void processProduces() {
+
+    if (produces == null) {
+      produces = parentContext.getProduces();
+    }
```
Comment (chanjarster) [inline oas-generator/oas-generator-core/src/main/java/org/apache/servicecomb/toolkit/generator/context/OperationContext.java]: Provide `getProduces()` method just like `getHttpMethod()` will be better.

File: oas-generator/oas-generator-spring/src/test/java/org/apache/servicecomb/toolkit/generator/SpringAnnotationProcessorTest.java [hunk 5]
```diff
@@ -55,13 +57,15 @@
 
   @Test
   public void getHttpMethod() throws NoSuchMethodException {
```
Comment (chanjarster) [inline oas-generator/oas-generator-spring/src/test/java/org/apache/servicecomb/toolkit/generator/SpringAnnotationProcessorTest.java]: This test is too big. Write individual method for each `@RequestMapping` property and a test for properties inheritance, eg, Class level `@RequestMapping` + Method level `RequestMapping`, will be clearer

**Comment by kakulisen** [comment]
I have optimized and submitted the code based on the review suggestions. 
Please help to check it again @chanjarster

**Pull request merged**